src/LINKS.hpp

```diff
@@ -283,80 +284,72 @@ class LINKS {
-    std::cout << "HASHES CALC: " << std::to_string(((double)m / bf_elements))
-              << " second: "
-              << std::to_string(((double)m / bf_elements) * log(2)) << "\n";
+
     unsigned hash_fct = floor(((double)m / bf_elements) * log(2));
-    std::cout << "- Number of bfElements: " << bf_elements << "\n"
-              << "- Input file path: " << links_arg_parser->bf_file << "\n"
-              << "- Input file: " << links_arg_parser->assembly_file << "\n"
-              << "- kmersize: " << links_arg_parser->k << "\n"
+    std::cout << "- Number of BF Elements: " << bf_elements << "\n"
+              << "- Input file path: " << links_arg_parser.base_name << "\n"
+              << "- Input file: " << links_arg_parser.assembly_file << "\n"
+              << "- kmersize: " << links_arg_parser.k << "\n"
               << "- m: " << m << "\n"
-              << "- fpr: " << links_arg_parser->fpr << "\n"
+              << "- fpr: " << links_arg_parser.fpr << "\n"
               << "- hashFct: " << hash_fct << "\n";
 
     std::string reading_tigbloom_message =
         "\n\n=>Reading contig/sequence assembly file : " +
         std::to_string(time(0)) + "\n";
 
-    std::cout << "- Filter output file : " << links_arg_parser->k << "\n";
+    std::cout << "- Filter output file : " << links_arg_parser.k << "\n";
     assembly_BF =
-        new btllib::KmerBloomFilter(m / 8, hash_fct, links_arg_parser->k);
-    btllib::SeqReader assembly_reader(links_arg_parser->assembly_file, 8, 1);
+        new btllib::KmerBloomFilter(m / 8, hash_fct, links_arg_parser.k);
+    btllib::SeqReader assembly_reader(links_arg_parser.assembly_file, 8, 1);
     size_t builder = 0;
     for (btllib::SeqReader::Record record; (record = assembly_reader.read());) {
 
       builder++;
       assembly_BF->insert(record.seq);
     }
     std::string bfmsg = "\n\nWriting Bloom filter to disk (" +
-                        links_arg_parser->bf_file +
-                        ") : " + std::to_string(time(0)) + "\n";
+                        links_arg_parser.base_name + ".bloom" + ")\n";
     std::cout << bfmsg;
-    assembly_BF->save(links_arg_parser->bf_file + ".bloom");
-    std::cout << "Done mybf, printing stats...\n";
+    assembly_BF->save(links_arg_parser.base_name + ".bloom");
   }
   return assembly_BF;
 }
 
-inline LINKS::LINKS(InputParser *input_parser)
-    : input_parser(input_parser), assembly_file(input_parser->assembly_file),
-      long_reads(input_parser->long_reads), distances(input_parser->distances),
-      k(input_parser->k), verbose(input_parser->verbose),
-      min_size(input_parser->min_size), step_sizes(input_parser->step_sizes),
-      insert_stdev(input_parser->insert_stdev),
-      base_name(input_parser->base_name), offset(input_parser->offset),
-      fpr(input_parser->fpr), bf_file(input_parser->bf_file),
-      bf_off(input_parser->bf_off), threads(input_parser->thread),
+inline LINKS::LINKS(InputParser input_parser)
```

Comment:
Why are you passing the InputParser object here if you're also using its values to construct members of the LINKS class?

Reply:
Hmm, what is the alternative? How can I access member fields of `InputParser` object without passing the object to `LINKS` object?

Reply:
Sorry, I misworded that. I wanted to ask, why does the LINKS class have an `InputParser input_parser` member that you pass in the constructor, if you use its values in the LINKS constructor, anyway?

Reply:
My recommendation would be, since there's a bunch of parameters/values that you need from `InputParser` to just initialize the `input_parser` member and use its members in LINKS instead of initializing LINKS members with the values from `InputParser`.

Reply:
That's fair. Deleted `InputParser` member from `LINKS`.